File: app/src/test/java/com/hades/example/android/qa/v1/AlertParamGetterTest3.java
package com.hades.example.android.qa.v1;

import static org.mockito.Mockito.mock;

import android.app.Activity;

import com.hades.example.android.qa.AlertParam;
import com.hades.example.android.qa.DefaultAlertConfigure;
import com.hades.example.android.qa.DefaultUIConfigure;

import org.junit.Assert;
import org.junit.Test;

public class AlertParamGetterTest3 {

    private final String defaultTitle = "Default title";
    private final String customTitle = "Custom title";

    @Test
    public void getAlertParam_1() { // ok
        DefaultUIConfigure configure = null;
        AlertParamGetter3 getter = new AlertParamGetter3(mock(Activity.class), configure) {
            public String getDefaultTitle() {
                return defaultTitle;
            }
        };
        AlertParam param = getter.getAlertParam();
        Assert.assertEquals(defaultTitle, param.getTitle());
    }

    @Test
    public void getAlertParam_2() { // ok
        DefaultUIConfigure configure = new DefaultUIConfigure();
        configure.setDefaultAlertConfigure(new DefaultAlertConfigure());
        configure.getDefaultAlertConfigure().setTitle(customTitle);

        AlertParamGetter3 getter = new AlertParamGetter3(mock(Activity.class), configure) {
            public String getDefaultTitle() {
                return defaultTitle;
            }
        };
        AlertParam param = getter.getAlertParam();
        Assert.assertEquals(customTitle, param.getTitle());
    }
}

File: app/src/test/java/com/hades/example/android/qa/v2/AlertParamGetterTest3.kt
package com.hades.example.android.qa.v2

import android.app.Activity
import com.hades.example.android.R
import com.hades.example.android.qa.DefaultAlertConfigure
import com.hades.example.android.qa.DefaultUIConfigure
import org.junit.Assert
import org.junit.Test
import org.mockito.Mockito

class AlertParamGetterTest3 {
    private val defaultTitle = "Default title"
    private val customTitle = "Custom title"

    @Test
    fun alertParam_1() { // ok
        val configure: DefaultUIConfigure? = null
        val getter: AlertParamGetter3 =
            object : AlertParamGetter3(Mockito.mock(Activity::class.java), configure) {
                override fun getDefaultTitle(): String? {
                    return defaultTitle
                }
            }
        val param = getter.getAlertParam()
        Assert.assertEquals(defaultTitle, param.title)
    }

    @Test
    fun alertParam_2() { // ok
        val configure = DefaultUIConfigure()
        configure.defaultAlertConfigure = DefaultAlertConfigure()
        configure.defaultAlertConfigure?.title = customTitle

        val getter: AlertParamGetter3 = object : AlertParamGetter3(
            Mockito.mock(
                Activity::class.java
            ), configure
        ) {
            override fun getDefaultTitle(): String? {
                return defaultTitle
            }
        }
        val param = getter.getAlertParam()
        Assert.assertEquals(customTitle, param.title)
    }
}